framework/include/constraints/EqualValueEmbeddedConstraint.h

```diff
@@ -54,9 +55,14 @@ class EqualValueEmbeddedConstraint : public NodeElemConstraint
   /// Formulations, currently only supports KINEMATIC and PENALTY
   const enum class Formulation { KINEMATIC, PENALTY } _formulation;
   /// Penalty parameter used in constraint enforcement for kinematic and penalty formulations
-  const Real _penalty;
+  const GenericReal<is_ad> _penalty;
```

Comment:
if ADEqual...Constaint does not work for kinematic, isnt this always a constant?

Reply:
penalty is constant.  NonAD will work for kinematic, AD won't.